### DISCUSSION TITLE
Reserved for System Use (and any variation on the theme)

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#142] [General]
How much of this was used for anything internally?  Without a DEEP dive, is there a simple answer?  I note the third category, "reserved for CDC use" (and variations thereof).

My take is that any under this category are basically an extension of "reserved for installation use", at least in 2024; indeed, "reserved for installation use" in most cases might want to be protected for compatibility with existing stuff, should it ever...  NOS is pretty well a fixed point.  Reserved doesn't mean what it did.

**Comment by wschaub** [comment]
If you look through the source code between versions you can see some of that reserved space eventually get carved out. The reserved for installation use stuff I don't think is used by any official products. and probably is the safest to use for things like site mods if you're into it. 

as far as 2.8.7 goes I do have my eyes on some of that reserved space for maybe handling time past 2033, problem with some of that reserved stuff is making sure if it's actually for future use or if it's being used and not documented.

**Reply by KC5VDJ** [reply]
I was thinking of a new pseudo-device for better authentication; and longer file passwords.  NOS 2.8.7 compat, unless set, then gets hash/whatever by that path.

Authentication always passed through device channels anyway, this would add another.

Just something I'm kicking around.  Bring it up to modern base security.  Hell, the year of the final update of NOS, the challenges were hot and old standards fell.  56 bit security went quickly (slightly modified DES ECB, aka /etc/passwd).  Password length needs expanded all around.

If I were to change NOS itself, the authentication and file passwords would be IMHO the best use of a bit of the unused(?) "system reserved" and "reserved".  I would be a bit apprehensive in a distributed NOS enhancement to use "installation reserved", so as not to stomp on someone's setup.

**Reply by KC5VDJ** [reply]
What I really need to find out is if using such words/bit-fields/bits stomps on NOS or if NOS would stomp on it?

Seeing "CDC reserved" seems to possibly indicate that "system reserved" in most cases might mean "reserved for future NOS versions"....in which case: safe(?)

**Reply by KC5VDJ** [reply]
The time thing is a problem should be dealt with like as soon as it can be done.  Like the *ix lineup did by going 64 bit with time, only old legacy stuff will need fixed, and it'll be a repeat of '99 again because of the legacy thing, unless dealt with early.  Granted, the money was good, and the jobs were interesting, but...not again, please.  GST/Armco was interesting (KCMO, next door to Independence, rod mill and melt shop, hp-ux/informix, and some mickeyvaxen (that scale gets in everything, but seems to be non-conductive, one 6300(?) had the better part of an inch of that inside, but was operating, had to clean it out to get to the board! (black dust)).

**Comment by kej715** [comment]
It's definitely risky to use a field that's specifically marked in
the source code as being reserved. In some cases, they might be changed as
side-effects of instructions. For example, a CWM instruction in a PP, or
SA6/SA7 instruction in the CPU, where the logic didn't take specific care
to preserve the CDC-reserved field across get/set. On the other hand, a
field specifically marked as being installation-available is definitely
fair game.
